torch/distributions.py

```diff
@@ -52,15 +52,30 @@ class Distribution(object):
     r"""
     Distribution is the abstract base class for probability distributions.
     """
+    has_sample_grad = False
```

Comment:
I’m wondering if `has_sample_grad` is a good name for it, since `sample` never has grad now. Maybe `has_rsample`?